store/tikv/gcworker/gc_worker.go

```diff
@@ -470,11 +477,16 @@ func resolveLocks(ctx goctx.Context, store tikv.Storage, safePoint uint64, ident
 			}
 			continue
 		}
-		regions++
+
 		totalResolvedLocks += len(locks)
-		key = loc.EndKey
-		if len(key) == 0 {
-			break
+		if len(locks) < gcScanLockLimit {
+			regions++
+			key = loc.EndKey
+			if len(key) == 0 {
+				break
+			}
+		} else {
+			key = locks[len(locks)-1].Key
```

Comment:
Should we change the key in this situtation? It seems we should still use the  current_region's start_key here?

Reply:
Yes, we should, if we don't set it here, and set the `start key` of `ScanLock` req to the key of end lock(we must), when the region split, we may find the `start key` is larger than the end key of the region, and we will get the error `key not in region`, then the gc job will exit this time.